### PR TITLE
chore: add sp price related cli and refine cli and comments

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

This PR adds cli related to storage provider price. Including editing and querying, and fix some existing cli usage. 

### Rationale

Enable the usage of cli.

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: x/sp/client/cli/tx.go [hunk 1]
```diff
@@ -715,3 +717,51 @@ Examples:
...
 	return cmd
 }
+
+func CmdUpdateStorageProviderStoragePrice() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-price [sp-address] [read-price] [store-price] [free-read-quota]",
+		Short: "Update prices and free read quota of a storage provider, all prices in BNB wei",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`update the storage provider read, store price and free read quota, if there is no change to a specific value, the current value should also be provided.
+Examples:
```
Comment (KeefeL) [inline x/sp/client/cli/tx.go:727]: I think we should describe more detail here, how to compute read/storage price, for example, if the expected storage price is 0.023$/GB/Month, then the read-price here should be 0.023*0.58/BNB Price*1e18/1024/1024/1024/30/24/60/60

Comment (KeefeL) [reply x/sp/client/cli/tx.go:727]: The free-read-quota unit is bytes, for 1GB free quota, it should be 1073741824.

File: x/sp/client/cli/tx.go [hunk 2]
```diff
@@ -715,3 +717,51 @@ Examples:
...
+ $ %s tx %s update-price 0x... 10000000 10000000 10000000
```
Comment (KeefeL) [inline x/sp/client/cli/tx.go:728]: 10000000 is not a good example, try to use the current mainnet configs here.

File: x/storage/types/tx.pb.go [hunk 3]
```diff
@@ -3150,7 +3150,7 @@ var fileDescriptor_ddb71b028305a3cc = []byte{
 	0x40, 0xd1, 0x21, 0xa6, 0xa5, 0x99, 0x0e, 0xea, 0x89, 0x7c, 0x53, 0x97, 0x35, 0x1f, 0xf6, 0xf4,
 	0x36, 0xf6, 0x7c, 0x9c, 0x1a, 0x2e, 0x61, 0xd5, 0x3f, 0xc1, 0xd4, 0x1e, 0x10, 0xcd, 0x17, 0x2a,
 	0xf8, 0x56, 0xbe, 0x0b, 0x40, 0x5d, 0xe4, 0x62, 0x66, 0x6a, 0x3f, 0x0b, 0xa7, 0x11, 0xdf, 0xf7,
-	0x81, 0xaa, 0xb4, 0x46, 0x69, 0xc7, 0x73, 0xe2, 0x6c, 0x66, 0x4e, 0x9c, 0x7b, 0xf2, 0x6c, 0x2d,
+	0x81, 0xaa, 0xb4, 0x46, 0x69, 0xc7, 0x73, 0xe2, 0x74, 0x66, 0x4e, 0x9c, 0x7b, 0xf2, 0x6c, 0x2d,
```
Comment (KeefeL) [inline x/storage/types/tx.pb.go:3153]: Isn't this a hardfork to modify this content?

File: x/storage/types/tx.pb.go [hunk 4]
```diff
@@ -5651,7 +5651,7 @@ func (m *MsgPutPolicy) MarshalToSizedBuffer(dAtA []byte) (int, error) {
 		i -= n6
 		i = encodeVarintTx(dAtA, i, uint64(n6))
 		i--
-		dAtA[i] = 0x3a
+		dAtA[i] = 0x2a
```
Comment (KeefeL) [inline x/storage/types/tx.pb.go:5654]: Not sure whether this is a hardfork, pelase help check @zjubfd

File: x/storage/types/tx.pb.go [hunk 5]
```diff
@@ -11675,7 +11675,7 @@ func (m *MsgPutPolicy) Unmarshal(dAtA []byte) error {
 				return err
 			}
 			iNdEx = postIndex
-		case 7:
+		case 5:
```
Comment (KeefeL) [inline x/storage/types/tx.pb.go:11678]: Also this one @zjubfd

**Review by zjubfd — CHANGES_REQUESTED** [review]
some proto type change may cause hardfork

File: x/sp/client/cli/tx.go (hunk 1 above)
Comment (alexgao001) [reply x/sp/client/cli/tx.go:734]: added

File: x/sp/client/cli/tx.go (hunk 2 above)
Comment (alexgao001) [reply x/sp/client/cli/tx.go]: fixed

File: x/storage/types/tx.pb.go (hunk 3 above)
Comment (alexgao001) [reply x/storage/types/tx.pb.go]: reverted

**Review by KeefeL — APPROVED** [review]

**Review by zjubfd — APPROVED** [review]

**Pull request merged**